spec/services/legacy/integration/data_importer_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+describe Legacy::DatasetImportService do
```

Comment:
This test suite may not be required - to be removed

Reply:
we should put everything in `dataset_import_service_spec`. All this stuff is unit testing, not integration.

Reply:
Currently 'unit/dataset_import_service_spec' contains both integration and unit tests. 

When you call the run method, I believe you are also invoking ActiveRecord and the db.

I could move the #run tests into 'integration/dataset_import_service_spec', or may worth considering mocking ActiveRecord